cpp/src/strings/regex/regex.cuh

```diff
@@ -25,6 +25,8 @@
 #include <thrust/optional.h>
 #include <thrust/pair.h>
 
+#include <cuda_runtime.h>
```

Comment:
If this change is to allow this file to included into cpp files, should we also move this file to an hpp? It's a semantic distinction for headers but it seems cleaner to mirror what we would do for source files (cu->cpp).

Reply:
No, because it includes device code at the end of the file. I will consider refactoring this in a later PR perhaps.